backend/store/stacktraces.go

```diff
@@ -228,19 +215,37 @@ func (store *Store) GitHubEnhancedStackTrace(ctx context.Context, stackTrace []*
 			continue
 		}
 
-		filesEnhanced += 1
+		// check if we've previously errored on this file
+		previousError, _ := store.redis.GetGitHubFileError(ctx, *service.GithubRepoPath, *validServiceVersion, fileName)
+		if previousError {
+			newMappedStackTrace = append(newMappedStackTrace, trace)
```

Comment:
nit: refactor this fn to return so that the `			newMappedStackTrace = append(newMappedStackTrace, trace)` logic can be reused, rather than copying it / continue everywhere we want to skip?

Reply:
Let me know if the changes in https://github.com/highlight/highlight/pull/6740/commits/85c88b41047f94dee63de2d1139db3cab9abd4ba

Definitely looks a little cleaner - only downside I see is a little confusing returning two booleans for the enhanceable and enhanced state, but tried to call out what these meant in a comment